batchflow/models/tf/deep_galerkin.py

```diff
@@ -0,0 +1,410 @@
+                rhs_val = rhs
+                rhs = lambda *args: rhs_val * tf.ones_like(tf.reduce_sum(points, axis=1, keepdims=True))
+            else:
+                raise ValueError("Cannot parse right-hand-side of the equation")
+
+        self.store_to_attr('targets', rhs(points))
+        return placeholders_, tensors_
+
+    @classmethod
+    def _make_form_calculator(cls, form, coordinates, name='_callable'):
+        """ Get callable that computes differential form of a tf.Tensor
+        with respect to coordinates.
+        """
+        n_dims = len(coordinates)
+        d1_coeffs = np.array(form.get("d1", np.zeros(shape=(n_dims, )))).reshape(-1)
+        d2_coeffs = np.array(form.get("d2", np.zeros(shape=(n_dims, n_dims)))).reshape(n_dims, n_dims)
+
+        if (np.all(d1_coeffs == 0) and np.all(d2_coeffs == 0)):
+            raise ValueError('Nothing to compute here! Either d1 or d2 must be non-zero')
+
+        def _callable(net):
+            """ Compute differential form.
+            """
+            # derivatives of the first order
+            vars = [coordinates[i] for i in np.nonzero(d1_coeffs)[0]]
+            result = sum(coeff * d1_ for coeff, d1_ in zip(d1_coeffs[d1_coeffs != 0], tf.gradients(net, vars)))
+
+            # derivatives of the second order
+            for i in range(n_dims):
+                vars = [coordinates[i] for i in np.nonzero(d2_coeffs[i, :])[0]]
+                if len(coordinates) > 0:
+                    d1_ = tf.gradients(net, coordinates[i])[0]
+                    result += sum(coeff * d2_ for coeff, d2_ in zip(d2_coeffs[i, d2_coeffs[i, :] != 0],
+                                                                    tf.gradients(d1_, vars)))
+            return result
+
+        setattr(_callable, '__name__', name)
+        return _callable
+
+    def _build(self, config=None):
+        """ Overloads :meth:`.TFModel._build`: adds ansatz-block for binding
+        boundary and initial conditions.
+        """
+        inputs = config.pop('initial_block/inputs')
+        x = self._add_block('initial_block', config, inputs=inputs)
+        x = self._add_block('body', config, inputs=x)
+        x = self._add_block('head', config, inputs=x)
+        output = self._add_block('ansatz', config, inputs=x)
+        self.store_to_attr('solution', output)
+        self.output(output, predictions=config['predictions'], ops=config['output'], **config['common'])
+
+    @classmethod
+    def ansatz(cls, inputs, **kwargs):
+        """ Binds `initial_condition` or `boundary_condition`, if these are supplied in the config
+        of the model. Does so by applying one of preset multipliers to the network output. Creates
+        a tf.Tensor `solution` - the final output of the model.
+        """
+        if kwargs.get("bind_bc_ic"):
+            add_term = 0
+            multiplier = 1
+
+            # retrieving variables
+            n_dims = kwargs.get('n_dims')
```

Comment:
Why do you use `.get`? In the next line it can be only `int`

Reply:
fixed